active/NumberBox/NumberBox.md

```diff
@@ -0,0 +1,220 @@
+# Number box
+
+<!-- The purpose of this spec is to describe a new feature and
+its APIs that make up a new feature in WinUI. -->
+
+<!-- There are two audiences for the spec. The first are people
+that want to evaluate and give feedback on the API, as part of
+the submission process.  When it's complete
+it will be incorporated into the public documentation at
+docs.microsoft.com (http://docs.microsoft.com/uwp/toolkits/winui/).
+Hopefully we'll be able to copy it mostly verbatim.
+So the second audience is everyone that reads there to learn how
+and why to use this API. -->
+
+## Background
+<!-- Use this section to provide background context for the new API(s) 
+in this spec. -->
+
+<!-- This section and the appendix are the only sections that likely
+do not get copied to docs.microsoft.com; they're just an aid to reading this spec. -->
+
+<!-- If you're modifying an existing API, included a link here to the
+existing page(s) -->
+
+<!-- For example, this section is a place to explain why you're adding this API rather than
+modifying an existing API. -->
+
+<!-- For example, this is a place to provide a brief explanation of some dependent
+area, just explanation enough to understand this new API, rather than telling
+the reader "go read 100 pages of background information posted at ...". -->
+
+
+## Description
+<!-- Use this section to provide a brief description of the feature.
+For an example, see the introduction to the PasswordBox control 
+(http://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box). -->
+
+Number box is a numerics only text control with support for input formatting, validation, basic calculation, and increment stepping.
```

Comment:
I think the use cases listed are too broad. Incrementing, decrementing and calculations target a very different set of use cases than IP addresses and phone numbers.

`NumericUpDown` is a classic WinForms control, later reimplemented by many in WPF and UWP for entering and updating values. It would be useful to have one in the platform and supporting calculations would be a somewhat useful if hidden and costly feature. Having one with a `double Value` property for binding to values would be very useful.

A `NumberBox` for phone numbers, IP numbers, SSNs and other ID codes with masking and formatting support would also be useful, but it sounds like a completely different use case. There is no single numeric value it would make sense for it to expose. It doesn't make sense to perform math operations or increment/decrement values of any of these data types.

Supporting currencies is yet another story. WinRT has no currency data type as far as I know and I doubt a `double` would suffice as a reliable money type even if you also specified a currency. It would be useful to have a `CurrencyBox` or `MoneyBox` control would be useful, but it would have to be much more specialized than a `NumberBox` or `NumericUpDown` could be.

Reply:
Agreed - the proposal has since been significantly scoped down to no longer support masked numerical strings - only numbers and math. 

We are still investigating how heavy inline calculation will be. I have reason to think it won't be terrible considering it is only intended to support basic math (*,-,/,+), operator precedence, and parenthetical operators but we will verify. 

Does this respond fully to everything you said?